### PR TITLE
stripe payments

### PR DESCRIPTION
**Pull request by carteryu**
tried to make it as ez as possible, let me know if u have any questions

looks like i have codys branch diffs too this should be cleaner when i diff to master

just use the <Stripe/> component from Stripe.js and u should be good to go

make sure aws-config and settings are up to date tho (ask me if u dont got)

### CODE REVIEW

File: RushDFS/src/Payments/PaymentFormView.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  render() {
+    const { onSubmit, submitted, error } = this.props
+    return (
+      <View style={{ flex: 1 }}>
+        <View>
+          <LiteCreditCardInput inputStyle={{ color: 'white' }} onChange={cardData => this.setState({ cardData })} />
+        </View>
+        <View style={styles.buttonWrapper}>
+          <Button
+            title='Submit to stripe'
```
Comment (EphemeralX) [inline RushDFS/src/Payments/PaymentFormView.js]: capitalize Stripe

File: RushDFS/src/Payments/PaymentFormView.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            //disabled={!this.state.cardData.valid || submitted} // logic for in case u wanna disable the button
+            onPress={() => {
+              console.log('card info:', this.state.cardData)
+              onSubmit(this.state.cardData)
+            }}
+          />
+          {/* TODO: On invalid card, i think we should have a component that renders instead of a new page but up to you */}
```
Comment (EphemeralX) [inline RushDFS/src/Payments/PaymentFormView.js]: unneeded, also you say new page, but I see an error component... ?

File: RushDFS/src/Payments/PaymentFormView.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    const { onSubmit, submitted, error } = this.props
+    return (
+      <View style={{ flex: 1 }}>
+        <View>
+          <LiteCreditCardInput inputStyle={{ color: 'white' }} onChange={cardData => this.setState({ cardData })} />
+        </View>
+        <View style={styles.buttonWrapper}>
+          <Button
+            title='Submit to stripe'
+            //disabled={!this.state.cardData.valid || submitted} // logic for in case u wanna disable the button
```
Comment (EphemeralX) [inline RushDFS/src/Payments/PaymentFormView.js]: get this question answered and then either have this code or leave it out pls. I lean towards disabling the button until the credit card is filled out.

File: RushDFS/src/Payments/Stripe.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      .map(key => key + '=' + card[key])
+      .join('&')
+  }).then(response => response.json())
+}
+
+const chargeUser = async stripeChargeRequest => {
+  await StripeAPI.chargeCard(stripeChargeRequest)
+}
+
+export default class Stripe extends React.Component {
```
Comment (EphemeralX) [inline RushDFS/src/Payments/Stripe.js]: what is the purpose of this class? just Stripe is hella ambiguous, especially when we have a StripeAPI above.  What's the overall code design/organization for the payments code? It's unclear to me looking at this PR.

File: RushDFS/src/Payments/Stripe.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    method: 'POST',
+    body: Object.keys(card)
+      .map(key => key + '=' + card[key])
+      .join('&')
+  }).then(response => response.json())
+}
+
+const chargeUser = async stripeChargeRequest => {
+  await StripeAPI.chargeCard(stripeChargeRequest)
+}
```
Comment (EphemeralX) [inline RushDFS/src/Payments/Stripe.js]: why do you have this outside the class?

Comment (EphemeralX) [inline RushDFS/src/Payments/Stripe.js]: what is chargeUser? are they depositing money into their app balance? are they buying something?  is this what needs to be done to allocate part of the balance on a parley? terminology is ambiguous, but important here.

File: RushDFS/src/Payments/Stripe.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+export default class Stripe extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      submitted: false,
+      error: null
+    }
+  }
+
+  onSubmit = async creditCardInput => {
```
Comment (EphemeralX) [inline RushDFS/src/Payments/Stripe.js]: this function does 3 things: 1. get credit card token. 2. make a charge request. 3. process charge request error if present.  we should split this into 3 functions with parameters from the previous step and shorten this long function.

File: RushDFS/src/graphql/mutations.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable */
+// this is an auto generated file. This will be overwritten
+
+export const chargeCard = `mutation ChargeCard($token: String!, $amount: String!, $currency: String!) {
+  chargeCard(token: $token, amount: $amount, currency: $currency)
+}
```
Comment (EphemeralX) [inline RushDFS/src/graphql/mutations.js]: are you just using this 1 API right now to test Stripe integration? realistically what actions/APIs should we expect?

Comment (carteryu) [reply RushDFS/src/graphql/mutations.js]: Its one of the mutations that we need

**Pull request closed without merging**